components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +263,86 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
-	certVolume := corev1.Volume{
-		Name: "trusted-ca",
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "trusted-ca",
-				},
-				Items: []corev1.KeyToPath{{
-					Key:  "ca-bundle.crt",
-					Path: "tls-ca-bundle.pem",
-				},
-				},
-			},
-		},
-	}
-	for index, volume := range *notebookVolumes {
-		if volume.Name == "trusted-ca" {
-			(*notebookVolumes)[index] = certVolume
-			certVolumeExists = true
-			break
-		}
-	}
-	if !certVolumeExists {
-		*notebookVolumes = append(*notebookVolumes, certVolume)
+	configMapName := "odh-trusted-ca-bundle"
+
+	// Fetch the list of ConfigMaps from the cluster
+	configMapList := &corev1.ConfigMapList{}
+	if err := cli.List(ctx, configMapList); err != nil {
+		return err
 	}
 
-	// Update Notebook Image container with env variables and Volume Mounts
-	for _, container := range *notebookContainers {
-		// Update notebook image container with env Variables
-		if container.Name == notebook.Name {
-			var newVars []corev1.EnvVar
-			imgContainer = container
-
-			for key, val := range proxyEnvVars {
-				keyExists := false
-				for _, env := range imgContainer.Env {
-					if key == env.Name {
-						keyExists = true
-						// Update if Proxy spec is updated
-						if env.Value != val {
-							env.Value = val
-						}
-					}
-				}
-				if !keyExists {
-					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
-				}
+	// Search for the odh-trusted-ca-bundle ConfigMap
+	for i := range configMapList.Items {
+		cm := &configMapList.Items[i]
+		if cm.Name == configMapName {
+
+			// Check if the odh-ca-bundle.crt key exists
+			certData, ok := cm.Data["odh-ca-bundle.crt"]
+			if !ok {
+				return fmt.Errorf("key 'odh-ca-bundle.crt' not found in ConfigMap '%s'", configMapName)
 			}
 
-			// Update container only when required env variables are not present
-			imgContainerExists := false
-			if len(newVars) != 0 {
-				imgContainer.Env = append(imgContainer.Env, newVars...)
+			// Attempt to decode PEM encoded certificate
+			block, _ := pem.Decode([]byte(certData))
+			if block != nil && block.Type == "CERTIFICATE" {
+				// Attempt to parse the certificate
+				_, err := x509.ParseCertificate(block.Bytes)
+				if err != nil {
+					return fmt.Errorf("error parsing certificate for key 'odh-ca-bundle.crt' in ConfigMap '%s': %v", configMapName, err)
+				}
+			} else if len(certData) > 0 {
+				return fmt.Errorf("invalid certificate format for key 'odh-ca-bundle.crt' in ConfigMap '%s'", configMapName)
 			}
 
-			// Create Volume mount
-			volumeMountExists := false
-			containerVolMounts := &imgContainer.VolumeMounts
-			trustedCAVolMount := corev1.VolumeMount{
-				Name:      "trusted-ca",
+			volumeName := "trusted-ca"
+			volumeMountPath := "/etc/pki/tls/certs/ca.crt"
+			volumeMount := corev1.VolumeMount{
+				Name:      volumeName,
+				MountPath: volumeMountPath,
+				SubPath:   "ca.crt",
```

Comment:
subpath value and the end value should match the path value
for example: `odh-ca-bundle.crt` specified on line 332

and also we should mount both crt, both:
`odh-ca-bundle.crt` and `ca-bundle.crt"`

Reply:
Thank you for pointing this, i had the intention that it mounted both crts in the previous implementation. Could you take a look on the new changes to check if this what we want?